internal/resolution/variable_sources/required_package/required_package_test.go

```diff
@@ -120,6 +122,33 @@ var _ = Describe("RequiredPackageVariableSource", func() {
 		}))
 	})
 
+	It("should filter by version range", func() {
+		// recreate source with version range option
+		var err error
+		rpvs, err = required_package.NewRequiredPackage(packageName, required_package.InVersionRange(">=1.0.0 !1.0.0 <3.0.0"))
```

Comment:
Is this valid range? i.e. include 1.0.0 then exclude it?

Reply:
That's a fair point. I think it's a valid range, even if it's superfluous (just collapses to >1.0.0 <3.0.0. I'll change it to !2.0.0 to avoid this confusion.